code/datums/records/record.dm

```diff
@@ -0,0 +1,615 @@
+						<th>Details</th>
+						<th>Author</th>
+						<th>Time Added</th>
+						</tr>"}
+	for(var/datum/crime_record/crime in crimes)
+		if(crime.valid)
+			final_paper_text += "<tr><td>[crime.name]</td>"
+			final_paper_text += "<td>[crime.details]</td>"
+			final_paper_text += "<td>[crime.author]</td>"
+			final_paper_text += "<td>[crime.time]</td>"
+		else
+			for(var/i in 1 to 4)
+				final_paper_text += "<td>--REDACTED--</td>"
+		final_paper_text += "</tr>"
+	final_paper_text += "</table><br><br>"
+
+	final_paper_text += "Citations:<br>"
+	final_paper_text  += {"<table style="text-align:center;" border="1" cellspacing="0" width="100%">
+						<tr>
+						<th>Citation</th>
+						<th>Details</th>
+						<th>Author</th>
+						<th>Time Added</th>
+						<th>Fine</th>
+						</tr><br>"}
+	for(var/datum/crime_record/citation/warrant in citations)
+		final_paper_text += "<tr><td>[warrant.name]</td>"
+		final_paper_text += "<td>[warrant.details]</td>"
+		final_paper_text += "<td>[warrant.author]</td>"
+		final_paper_text += "<td>[warrant.time]</td>"
+		final_paper_text += "<td>[warrant.fine]</td>"
+		final_paper_text += "</tr>"
+	final_paper_text += "</table><br><br>"
+
+	final_paper_text += text("<center>Important Notes:</center><br>")
+	if(security_note)
+		final_paper_text += text("- [security_note]<br>")
+	if(description)
+		final_paper_text += text("- [description]<br>")
+
+	printed_paper.name = text("SR-[] '[]'", print_count, name)
+	printed_paper.add_raw_text(final_paper_text)
+	printed_paper.update_appearance()
+
+	return printed_paper
+
+
+/// Returns a paper printout of the current record's crime data.
+/datum/record/crew/proc/get_medical_sheet()
+	var/obj/item/paper/med_record_paper = new
+	var/med_record_text
+	med_record_text += "<CENTER><B>Medical Record</B></CENTER>"
+	med_record_text += "<BR>Name: [name] Rank: [rank]"
+	med_record_text += "<BR>Gender: [gender]</BR>"
+	med_record_text += "<BR>Age: [age]<BR>"
+	med_record_text += "<BR>\n<CENTER><B>Medical Data</B></CENTER></BR>"
+	med_record_text += "<BR>Blood Type: [blood_type]</BR>"
+	med_record_text += "<BR>DNA: [dna_string]</BR>"
+	med_record_text += "<BR>Physical Status: [physical_status]</BR>"
+	med_record_text += "<BR>Mental Status: [mental_status]</BR>"
+	med_record_text += "<BR>Minor Disabilities: [minor_disabilities]</BR>"
+	med_record_text += "<BR>Details: [minor_disabilities_desc]</BR>"
+	med_record_text += "<BR>Major Disabilities: [major_disabilities]</BR>"
+	med_record_text += "<BR>Details: [major_disabilities_desc]</BR>"
+	med_record_text += "<BR>Important Notes: \t[medical_notes]</BR>"
+	med_record_text += "<BR><CENTER><B>Comments/Log</B></CENTER></BR>"
+
+	med_record_paper.name = "paper - '[name]'"
+	med_record_paper.add_raw_text(med_record_text)
+	med_record_paper.update_appearance()
+
+	return med_record_paper
+
+/**
+ * Cloning record
+ */
+/datum/record/cloning
+
+	var/id
+	var/datum/dna/dna
+	var/uni_identity
+	var/SE
+	var/datum/mind/mind /// Mind datum
```

Comment:
taking dna/mind instance directly is risky for harddel issue.
This should be...
```suggestion
	var/datum/weakref/weakref_dna
	var/uni_identity
	var/SE
	var/datum/weakref/weakref_mind /// Mind datum
```